lib/transfer.cpp

```diff
@@ -79,12 +79,26 @@ namespace quda {
       total_block_size *= geo_bs[d];
     }
 
-    if (total_block_size == 1) errorQuda("Total geometric block size is 1");
+    // Aggregation size is "technically" 1 for optimized KD
+    if (total_block_size == 1 && transfer_type != QUDA_TRANSFER_OPTIMIZED_KD) errorQuda("Total geometric block size is 1 but transfer type isn't optimized-kd");
```

Comment:
As noted elsewhere, we want to preserve the ability to thin the degrees of freedom at constant geometry without needing KD optimized build.

Reply:
Fixed via logic noted in other comment, `if (total_block_size != 1 && transfer_type == QUDA_TRANSFER_OPTIMIZED_KD)`. Updated the error appropriately to note that geometric block size must be 1 for optimized KD op.